### DISCUSSION TITLE
experiment reproduction using lighthouse positioning system

### DISCUSSION THREAD

**Discussion by an949na** [TinyMPC/TinyMPC#46] [Q&A] [answered]
Hi! I'm trying to reproduce your experiment video with using Lighthouse Positioning System. But there is so many problems about this reproduction. I finished localization perfectly with Lighthouse Positioning System. I want to know how to put TinyMPC Firmware with Lighthouse Positioning System. Please help me!

**Comment by xkhainguyen** [comment]
Please take a look at #40

**Comment by an949na** [comment]
Thank you for your fast response. I checked #40 session, but  I didn't send
the email because it felt rude to do so without any prior notice. Sorry for
bothering you.

This is my problem. I want to import TinyMPC in my crazyflie 2.1, using the
Lighthouse Positioning System. I've checked your TinyMPC Github.
Unfortunately, I think my coding skill is not enough to import new firmware
in my drone. Is there any helpful advice or record section that can help
me? I'm looking forward to hearing from you. Thank you!

2024년 11월 27일 (수) 오후 5:52, Khai Nguyen ***@***.***>님이 작성:

**Comment by xkhainguyen** [comment]
1. [Python interface example](https://github.com/TinyMPC/tinympc-python/blob/113698875f717a5d275f7e23dd72cdde858074ba/examples/interactive_quadrotor_ext.ipynb) should cover most of the building blocks for prototyping TinyMPC on the Crazyflie—this workflow should be similar for your drone as well.
2. Our [Julia](https://github.com/RoboticExplorationLab/tinympc-dev-julia/tree/khai/dev/quat_mpc_tuning) test experiments.
3. The [firmware](https://github.com/RoboticExplorationLab/tinympc-crazyflie-firmware) which you will be able to flash successfully..
4. Crazyswarm [scripts](https://github.com/sschoedel/crazyswarm/tree/demo/socp) for communication.

I sent you the unofficial tutorial videos via email. Let me know if you have any questions.

**Comment by xkhainguyen** [comment, marked as answer]
TinyMPC can be used with any kind of positioning system (we used MoCap or Flow deck). People have successfully used TinyMPC with the Lighthouse system. 

Please take a look at this: [https://magperch.mobirisesite.com/ ](https://magperch.mobirisesite.com/ )
You can reach out to the site owner for more information.
